### ISSUE TITLE
"\M+5D7DF\M+5CFDF\M+5BCDC" is invalid value for layer name

### ISSUE THREAD

**Issue by maslke** [mozman/ezdxf#932]
**Describe the bug**

I have some DWG files, and after converting them to DXF files using ODAFileConverter, I found that some of the layer names are strings like "\M+5D7DF\M+5CFDF\M+5BCDC".
When attempting to read these DXF files using ezdxf, an error occurred.
The error message is as follows:
```

Traceback (most recent call last):
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/addons/drawing/qtviewer.py", line 256, in _select_doc
    doc, auditor = recover.readfile(path)
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/recover.py", line 80, in readfile
    doc, auditor = read(fp, errors=errors)
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/recover.py", line 105, in read
    return _load_and_audit_document(recover_tool)
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/recover.py", line 142, in _load_and_audit_document
    doc._load_section_dict(recover_tool.section_dict)
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/document.py", line 394, in _load_section_dict
    loader.load_and_bind_dxf_content(sections, self)
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/lldxf/loader.py", line 156, in load_and_bind_dxf_content
    factory.bind(entity, doc)
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/entities/factory.py", line 88, in bind
    doc.entitydb.add(entity)
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/entitydb.py", line 173, in add
    entity.add_sub_entities_to_entitydb(self)  # type: ignore
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/entities/subentity.py", line 80, in add_sub_entities_to_entitydb
    self.new_seqend()
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/entities/subentity.py", line 87, in new_seqend
    seqend = factory.create_db_entry("SEQEND", attribs, self.doc)
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/entities/factory.py", line 63, in create_db_entry
    entity = new(dxftype=dxftype, dxfattribs=dxfattribs)
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/entities/factory.py", line 53, in new
    entity = cls(dxftype).new(
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/entities/dxfentity.py", line 166, in new
    entity.update_dxf_attribs(attribs)
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/entities/dxfentity.py", line 448, in update_dxf_attribs
    setter(key, value)
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/entities/dxfns.py", line 230, in set
    self.__setattr__(key, value)
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/entities/dxfns.py", line 172, in __setattr__
    self.__dict__[key] = check(value)
  File "/home/maslke/.local/lib/python3.10/site-packages/ezdxf/entities/dxfns.py", line 161, in check
    raise const.DXFValueError(
ezdxf.lldxf.const.DXFValueError: Invalid value \M+5D7DF\M+5CFDF\M+5BCDC for attribute "layer" in entity SEQEND(#None).
```
I have tracked the code, and found codes in validator.py:
```
def is_valid_table_name(name: str) -> bool:
    # remove backslash of DXF unicode encoding \U+xxxx
    chars = set(name.replace(r"\U+", ""))
    return not bool(INVALID_LAYER_NAME_CHARACTERS.intersection(chars))
```

In above code, removes \U+ , but \U+ is not be removed.

If I edited is_valid_table_name like this, this dxf file will be read successfully.
```
def is_valid_table_name(name: str) -> bool:
    # remove backslash of DXF unicode encoding \U+xxxx
    chars = set(name.replace(r"\U+", "").replace(r"\M+", ""))
    return not bool(INVALID_LAYER_NAME_CHARACTERS.intersection(chars))
```

So，\M+ should be removed? Or, it is just a issue with my dxf fie and it's not a common issue.

**Comment by mozman** [comment]
The `"\M+"` encoding (what ever this is) is not documented and not supported by `ezdxf`.

**Comment by mozman** [comment]
Please post an zipped example DXF file for testing.

**Comment by maslke** [comment]
> Please post an zipped example DXF file for testing.

[test.zip](https://github.com/mozman/ezdxf/files/12674700/test.zip)

Using this dxf file can reproduce this issue.

PS: 
1. When I converted dwg file to dxf file using ODAFileConverter(version:24.8) in ubuntu 22.04 OS, then loaded by ezdxf, the problem exists.
2. When I converted dwg file to dxf file using ODAFileConverter(version:24.8) in Windows 11 OS, then loaded by ezdxf, the problem does not exists.

In 1 and 2, I used same dwf file.

Maybe the issue was  because of some fonts files missing in Ubuntu?

Here is the difference between 2 dxg file.

1. dxf file in Ubuntu22.04
```code
  9
$CLAYER
  8
\M+5D7DF\M+5CFDF\M+5BCDC
  9
$CELTYPE
```
2. dxf file in Windows 11
```
  9
$CLAYER
  8
走线架
 9
$CELTYPE
```

Thanks for your reply.

**Comment by mozman** [comment]
FYI: #859 is a related discussion, without a solution

**Comment by mozman** [comment]
Even ChatGPT can't help - I am not sure if ChatGPT knows anything about the "\M+xxxxx" encoding:

Translating the hexadecimal number in the "\M+" encoding to Unicode can be a bit tricky because it's a proprietary encoding specific to DXF (Drawing Exchange Format) files. The "\M+" encoding doesn't have a direct one-to-one mapping to Unicode code points, so you would need a DXF-specific decoder to interpret these codes properly.

If you have a DXF file that uses the "\M+" encoding and you want to convert it to Unicode, you may need to use a DXF parsing library or tool that can handle this encoding. These tools are designed to read DXF files and convert their content to a more usable format, such as plain text or other graphic formats. Some CAD software programs or third-party libraries may provide functionality to perform this kind of conversion.

In summary, translating "\M+" encoded characters to Unicode typically requires a DXF-specific parser or converter because it's not a standard Unicode encoding, and the mapping of "\M+" codes to Unicode code points may depend on the specific context within the DXF file.

**Comment by mbway** [comment]
I have found some information on this encoding. I searched for `M+` in the libredwg repository and found
- [documentation on text encoding](https://github.com/LibreDWG/libredwg/blob/9d42f75345e30d9f98129f1fe54b4e4cf6c8c7b5/doc/LibreDWG.texi#L1200)
- [an implementation to convert M+ to U+](https://github.com/LibreDWG/libredwg/blob/9d42f75345e30d9f98129f1fe54b4e4cf6c8c7b5/src/out_dxf.c#L1158)

**Comment by mozman** [comment]
So `ezdxf` can now decode the MIF notation when the DXF document is loaded via the `recover` module. This is the same behavior as for the `\U+xxxx` encoding. When saving by `ezdxf` the content is `utf-8` encoded for DXF R2007+ or `\U+xxxx` encoded for older DXF versions.

![mif_decoded](https://github.com/mozman/ezdxf/assets/4230966/ae283306-96e8-4298-ab26-feb1ed909764)
